prime-router/src/main/kotlin/cli/FileUtilities.kt

```diff
@@ -85,13 +81,18 @@ class FileUtilities {
             val outputFile = if (outputFileName != null) {
                 File(outputFileName)
             } else {
+                // is this config HL7?
+                val hl7Config = report.destination?.translation as? Hl7Configuration?
+                // if it is, get the test processing mode
+                val processingMode = hl7Config?.processingModeCode ?: "P"
                 val fileName = Report.formFilename(
```

Comment:
Looks better. Thanks. Could this use the new Report method that you just defined?